### PR TITLE
Mark as_view_of_rank_n as KOKKOS_FUNCTION

### PR DESCRIPTION
**Pull request by masterleinad**
We are using this function in other `KOKKOS_FUNCTION` functions.

### CODE REVIEW

**Review by PhilMiller — APPROVED** [review]
Thanks. Sorry I missed this when I added it

**Review by crtrott — APPROVED** [review]
Approving, come to think of that however it feels like the specialization of `as_view_of_rank_n`  for normal Views probably should move to Kokkos_DynRankView too.

**Review by dalg24 — DISMISSED** [review]

File: containers/src/Kokkos_DynRankView.hpp [hunk 1]
```diff
@@ -1701,11 +1701,15 @@ namespace Impl {
    underlying memory, to facilitate implementation of deep_copy() and
    other routines that are defined on View */
 template <unsigned N, typename T, typename... Args>
-auto as_view_of_rank_n(DynRankView<T, Args...> v) {
+KOKKOS_FUNCTION auto as_view_of_rank_n(DynRankView<T, Args...> v) {
   if (v.rank() != N) {
-    Kokkos::Impl::throw_runtime_exception(
```
[the remark this answers is not in the document]
Comment (dalg24) [reply containers/src/Kokkos_DynRankView.hpp:1706]: Death test is an option but doing it on the device gets annoying pretty quickly considering the current support in various backends

**Comment by dalg24** [comment]
Failure are unrelated (machine ran out of disk space)

**Pull request merged**